staking/programs/staking/src/lib.rs

```diff
@@ -585,11 +654,43 @@ pub mod staking {
             )?;
         }
 
+        ctx.accounts.source_stake_account_custody.reload()?;
+        ctx.accounts.new_stake_account_custody.reload()?;
+
+
+        // Post-check
+        utils::risk::validate(
```

Comment:
How do you feel about this checks, are we scared about them blocking a split request forever?

Reply:
I think we should keep them -- I would rather have a split request blocked forever than some stake accounts that violate the invariants. (I don't think either of those things will happen though)